named_test.go

```diff
@@ -296,3 +298,88 @@ func TestNamedQueries(t *testing.T) {
 
 	})
 }
+
+func TestNamedBulkInsert(t *testing.T) {
+	type Val struct {
+		K string `db:"k"`
+		V int    `db:"v"`
+	}
+
+	vs := []Val{{K: "x"}, {K: "y"}, {K: "z"}}
+	table := []struct {
+		values      []interface{}
+		q, expected string
+	}{
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k) VALUES (:k)",
+			expected: "INSERT INTO val (k) VALUES (?)",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k) VALUES (:k)",
+			expected: "INSERT INTO val (k) VALUES (?),(?)",
+		},
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k,v) VALUES (:k,:v)",
+			expected: "INSERT INTO val (k,v) VALUES (?,?)",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k,v) VALUES (:k,:v)",
+			expected: "INSERT INTO val (k,v) VALUES (?,?),(?,?)",
+		},
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k,v) VALUES ( :k, :v )",
+			expected: "INSERT INTO val (k,v) VALUES ( ?, ? )",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k,v) VALUES ( :k, :v )",
+			expected: "INSERT INTO val (k,v) VALUES ( ?, ? ),( ?, ? )",
+		},
+		// sql functions (0 arguments)
+		{
+			values: []interface{}{vs[0], vs[1]},
+			q: func() string {
+				_, _, now := defaultSchema.Postgres()
+				return fmt.Sprintf("INSERT INTO val (k, v, added_at) VALUES (:k, :v, %v)", now)
+			}(),
+			expected: "INSERT INTO val (k, v, added_at) VALUES (?, ?, now()),(?, ?, now())",
+		},
+		{
+			values: []interface{}{vs[0], vs[1]},
+			q: func() string {
+				_, _, now := defaultSchema.MySQL()
+				return fmt.Sprintf("INSERT INTO val (k, v, added_at) VALUES (:k, :v, %v)", now)
+			}(),
+			expected: "INSERT INTO val (k, v, added_at) VALUES (?, ?, now()),(?, ?, now())",
+		},
+		{
+			values: []interface{}{vs[0], vs[1]},
+			q: func() string {
+				_, _, now := defaultSchema.Sqlite3()
+				return fmt.Sprintf("INSERT INTO val (k, v, added_at) VALUES (:k, :v, %v)", now)
+			}(),
+			expected: "INSERT INTO val (k, v, added_at) VALUES (?, ?, CURRENT_TIMESTAMP),(?, ?, CURRENT_TIMESTAMP)",
+		},
+		// extra operation
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k,v) VALUES (:k,:v) ON DUPLICATE KEY UPDATE v = VALUES(v)",
```

Comment:
For this, I changed the regex values ​​to start with "VALUES".

Reply:
( But when this feature is used other than bulk insert (e.g. used in NamedQuery), this changes is broken )